bin/node/runtime/src/lib.rs

```diff
@@ -482,6 +477,7 @@ parameter_types! {
 	pub const SlashDeferDuration: pallet_staking::EraIndex = 24 * 7; // 1/4 the bonding duration.
 	pub const RewardCurve: &'static PiecewiseLinear<'static> = &REWARD_CURVE;
 	pub const MaxNominatorRewardedPerValidator: u32 = 256;
+	pub const OffendingValidatorsThreshold: Perbill = Perbill::from_percent(17);
```

Comment:
Just out of curiosity, how do we come up with this number? And why not ~33?

Reply:
By 33% finality would already be stalled so it would be a bit too late to take action. I didn't pick this number myself but I think the logic is to act when we're halfway through stalling consensus.